erpnext/accounts/report/statement_of_account/statement_of_account.js

```diff
@@ -99,7 +105,27 @@ frappe.query_reports["Statement of Account"] = {
 			"fieldtype": "Data",
 			"hidden": 1
 		},
-	]
+	],
 };
 
 erpnext.utils.add_dimensions('Statement of Account', 15)
+
+function get_addresses(report) {
+	let filters = report.get_filter_values();
+
+	if (!filters.company || !filters.party_type || !filters.party[0]) {
+		return
+	}
+
+	frappe.call({
+		method: "erpnext.accounts.report.statement_of_account.statement_of_account.get_addresses",
+		args: {
+			company: filters.company,
+			party_type: filters.party_type,
+			party: filters.party[0]
```

Comment:
Talk to @vishdha and Arundhati about this, maybe we don't want to allow multiple parties in that field.